activesupport/lib/active_support/railtie.rb

```diff
@@ -10,6 +10,8 @@ class Railtie < Rails::Railtie # :nodoc:
 
     config.eager_load_namespaces << ActiveSupport
 
+    ActiveSupport::LazyLoadHooks.warn_on_premature_loading = true
```

Comment:
How are users going to configure this? If they aren't we should not make it a configuration.

Reply:
I've removed this setting.